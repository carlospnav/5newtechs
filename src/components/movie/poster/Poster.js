import styled from "styled-components"
import { flexGroup } from "theme/utils"

const Poster = styled.div`
  width: 60%;
  ${flexGroup(null, "center")};

  @media (min-width: 1000px){
    width: 40%;
    justify-content: flex-start;
  }
`

export default function({episode}){
  return (
    <Poster>
      <img src={`/static/poster/episode${episode}.jpg`} alt="Movie Poster" />
    </Poster>
  )
}