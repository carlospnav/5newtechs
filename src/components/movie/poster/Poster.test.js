import { shallow, mount } from "enzyme"
import Poster from "./Poster"
import "jest-styled-components"

describe("<Poster>", () => {

  it("should render a Poster properly.", () => {
    const wrapper = shallow(<Poster episode="3" />)
    expect(wrapper.dive()).toMatchSnapshot()
  })

  /* Still debating whether or not to test this. Snapshot diffs should
  indicate if src is being set properly with the episode prop. Decided to 
  write this test for the sake of being thorough. */
  it("should receive episode 3 and pass it on to the img src.", () => {
    const wrapper = mount(<Poster episode="3" />)
    expect(`/static/poster/episode${wrapper.prop('episode')}.jpg`).toEqual(wrapper.find("img").prop("src"))
  })
})