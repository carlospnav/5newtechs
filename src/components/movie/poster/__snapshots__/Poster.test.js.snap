// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster> should render a Poster properly. 1`] = `
.c0 {
  width: 60%;
  -webkit-flex-direction: null;
  -ms-flex-direction: null;
  flex-direction: null;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:1000px) {
  .c0 {
    width: 40%;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div
  className="c0"
>
  <img
    alt="Movie Poster"
    src="/static/poster/episode3.jpg"
  />
</div>
`;
